src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -379,7 +403,10 @@ public static SchemaDefinition Create(Type userType, Direction direction = Direc
                 {
                     if (!KeyType.IsValidDataType(dataType))
                         throw Contracts.ExceptParam(nameof(userType), "Member {0} marked with KeyType attribute, but does not appear to be a valid kind of data for a key type", memberInfo.Name);
-                    itemType = new KeyType(dataType, keyAttr.Count);
+                    if (keyAttr.Count == null)
+                        itemType = new KeyType(dataType, dataType.ToMaxInt());
+                    else
+                        itemType = new KeyType(dataType, keyAttr.Count.GetValueOrDefault());
```

Comment:
Instead of rewriting the logic here, I would suggest using `KeyCount` and `ConstructKeyType` in TypeParsingUtils.cs as those are the objects and methods used to specify KeyTypes.

Reply:
ConstructKeyType  uses `DataKind` not type. But I like idea of KeyCount

---
In reply to: [258663366](https://github.com/dotnet/machinelearning/pull/2638#discussion_r258663366) [](ancestors = 258663366)

Reply:
Looks good! I guess you are right. If you feel like it you can extract that logic out of this method and `ConstructKeyType` but not sure it's worth it for four lines. 

---
In reply to: [258671672](https://github.com/dotnet/machinelearning/pull/2638#discussion_r258671672) [](ancestors = 258671672,258663366)